### PR TITLE
[extractor/clyp] Added wav download support

### PR DESCRIPTION
**Pull request by qulaz**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This PR added support to download wav's from clyp.it

Clyp API not returned links to wav's (only mp3 and ogg), but in html page exists link to wav file. I'm added this to extractor.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/clyp.py [hunk 1]
```diff
@@ -9,16 +9,16 @@
 class ClypIE(InfoExtractor):
     _VALID_URL = r'https?://(?:www\.)?clyp\.it/(?P<id>[a-z0-9]+)'
     _TESTS = [{
-        'url': 'https://clyp.it/ojz2wfah',
```
Comment (qulaz) [inline yt_dlp/extractor/clyp.py:12]: this test case is outdated, file was deleted

File: yt_dlp/extractor/clyp.py [hunk 2]
```diff
@@ -74,3 +97,11 @@ def _real_extract(self, url):
...
             'formats': formats,
         }
+
+    def get_wav_format(self, url, audio_id):
+        page = self._download_webpage(url, video_id=audio_id)
+        wav_url = self._html_search_regex(
+            r'var wavStreamUrl = \"(?P<url>.+)\"', page, 'url', default='')
+
+        if wav_url.startswith('https://'):
+            return wav_url
```
Comment (pukkandan) [inline yt_dlp/extractor/clyp.py:107]: * Doesn't need to be a function
* The https check can be part of the regex
* Relax the regex with `\s*`
* Change `\"` to `\'"`, `.+` to `[^\'"]+`

File: yt_dlp/extractor/clyp.py [hunk 3]
```diff
@@ -34,6 +34,21 @@ class ClypIE(InfoExtractor):
...
+            'ext': 'wav',
+            'title': 'i dont wanna go (old version)',
+            'description': None,
+            'duration': 113.528,
+            'timestamp': 1607348505,
+            'upload_date': '20201207',
+        },
+        'params': {
+            'format': 'wavStreamUrl',
+        },
```
Comment (pukkandan) [inline yt_dlp/extractor/clyp.py:51]:
```suggestion
```
Shouldn't be needed

File: yt_dlp/extractor/clyp.py [hunk 4]
```diff
@@ -34,6 +34,21 @@ class ClypIE(InfoExtractor):
...
             'skip_download': True,
         },
+    }, {
+        'url': 'https://clyp.it/v42214lc',
+        'md5': '4aca4dfc3236fb6d6ddc4ea08314f33f',
+        'info_dict': {
+            'id': 'v42214lc',
+            'ext': 'wav',
+            'title': 'i dont wanna go (old version)',
+            'description': None,
```
Comment (pukkandan) [inline yt_dlp/extractor/clyp.py:44]:
```suggestion
```

File: yt_dlp/extractor/clyp.py [hunk 5]
```diff
@@ -9,16 +9,16 @@
...
-            'id': 'ojz2wfah',
+            'id': '2poken35',
             'ext': 'mp3',
-            'title': 'Krisson80 - bits wip wip',
-            'description': '#Krisson80BitsWipWip #chiptune\n#wip',
-            'duration': 263.21,
-            'timestamp': 1443515251,
-            'upload_date': '20150929',
+            'title': 'research',
+            'description': None,
```
Comment (pukkandan) [inline yt_dlp/extractor/clyp.py:18]:
```suggestion
```
Neither of the tests have description? Add one with it is possible

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/clyp.py (hunk 3 above)
Comment (qulaz) [reply yt_dlp/extractor/clyp.py:51]: without this param, for some reason, mp3 is selected

**Comment by pukkandan** [comment]
Pls avoid force-pushing in middle of review

File: yt_dlp/extractor/clyp.py (hunk 3 above)
Comment (pukkandan) [reply yt_dlp/extractor/clyp.py:51]: Then extractor isn't working properly. Try adding `'ext': 'wav'` to the returned format

File: yt_dlp/extractor/clyp.py [hunk 6]
```diff
@@ -61,6 +76,16 @@ def _real_extract(self, url):
                         'vcodec': 'none',
                     })
 
+        page = self._download_webpage(url, video_id=audio_id)
+        wav_url = self._html_search_regex(
+            r'var\s*wavStreamUrl\s*=\s*["\'](?P<url>https?://[^\'"]+)', page, 'url', default='')
```
Comment (pukkandan) [inline yt_dlp/extractor/clyp.py:81]:
```suggestion
            r'var\s*wavStreamUrl\s*=\s*["\'](?P<url>https?://[^\'"]+)', page, 'url', default=None)
```

File: yt_dlp/extractor/clyp.py (hunk 3 above)
Comment (qulaz) [reply yt_dlp/extractor/clyp.py]: with `ext` result is the same, but with `acodec` it works correct

**Pull request merged**